### PR TITLE
Refactor DB interfaces and add s3 sync foundation

### PR DESCRIPTION
**Pull request by bsquizz**
Much of the file create/delete logic was being done within the API handlers.

The s3 sync will re-use a lot of the same logic, so this PR moves that logic into a separate location that both the API and the s3 sync will be able to use.

In addition, the "skeleton" of the s3 sync logic is added in this PR. We are going to have an agent configuration defined with a pretty simple YAML file. The config file will define your desired agents. For each agent, you will specify the name/description/system prompt and then the s3 bucket name, s3 bucket prefix (e.g.: `default/component/myapp`), and file extension types that we want to sync from that prefix in s3.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -30,7 +31,7 @@
         db.create_all()
```
Comment (maknop) [inline main.py:31]: I don't see a function definition for this but I do see `create`.

**Review by maknop — COMMENTED** [review]
Refactor looks good! Just had one question before I approve.

File: main.py (hunk 1 above)
Comment (maknop) [reply main.py:31]: I also don't see a function declaration for `init_app`. Maybe I'm missing something though.

Comment (bsquizz) [reply main.py:31]: This is flask-sqlalchemy initialization stuff. [See here](https://flask-sqlalchemy.palletsprojects.com/en/3.0.x/quickstart/)

So `db` is an instance of the `SQLAlchemy` class (see in common.py I call `db = SQLAlchemy()`, but `Agent` is a class inheriting from `db.Model` and it represents an agent in the database. The way I wrote the Agent class is based a bit [on this](https://github.com/cookiecutter-flask/cookiecutter-flask/blob/master/%7B%7Bcookiecutter.app_name%7D%7D/%7B%7Bcookiecutter.app_name%7D%7D/database.py#L15
)

Comment (maknop) [reply main.py:31]: Ohhh sorry, I missed that.

**Comment by maknop** [comment]
LGTM!

**Review by maknop — APPROVED** [review]

**Pull request merged**